File: Python3/프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.py
from functools import cmp_to_key

gn = 0
def sortByNth(a, b):
    if a[gn] != b[gn]:
        return -1 if a[gn] < b[gn] else 1
    return -1 if a < b else 1

def solution(strings, n):
    global gn
    gn = n
    return sorted(strings, key=cmp_to_key(sortByNth))

File: C++/프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int n;

bool compare(string a, string b) {
    if(a[n] == b[n])
        return a < b;
    return a[n] < b[n];
}

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    ::n = n;
    sort(strings.begin(), strings.end(), compare);
    return strings;
}